src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.cpp

```diff
@@ -314,165 +314,112 @@ void addDimensionToPack(ConversionPatternRewriter &rewriter, Location loc,
-        // TODO (tungld): update the following statement once our llvm-project
-        // is upgraded to include the patch.
-        auto isSigned = true;
-        if (width == 8) {
-          if (isSigned) {
-            int8_t value = std::numeric_limits<int8_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          } else {
-            uint8_t value = std::numeric_limits<uint8_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          }
-        } else if (width == 16) {
-          if (isSigned) {
-            int16_t value = std::numeric_limits<int16_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          } else {
-            uint16_t value = std::numeric_limits<uint16_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          }
-        } else if (width == 32) {
-          if (isSigned) {
-            int32_t value = std::numeric_limits<int32_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          } else {
-            uint32_t value = std::numeric_limits<uint32_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          }
-        } else if (width == 64) {
-          if (isSigned) {
-            int64_t value = std::numeric_limits<int64_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          } else {
-            uint64_t value = std::numeric_limits<uint64_t>::max();
-            constantAttr = rewriter.getIntegerAttr(type, APInt(width, value));
-          }
-        } else {
+      .Case<IntegerType>([&](IntegerType type) {
+        size_t width = type.getWidth();
+        bool isSigned = type.isSigned();
+        switch (width) {
+        case 8:
+          value = (isSigned) ? std::numeric_limits<int8_t>::max()
+                             : std::numeric_limits<uint8_t>::max();
+          break;
+        case 16:
+          value = (isSigned) ? std::numeric_limits<int16_t>::max()
+                             : std::numeric_limits<uint16_t>::max();
+          break;
+        case 32:
+          value = (isSigned) ? std::numeric_limits<int32_t>::max()
+                             : std::numeric_limits<uint32_t>::max();
+          break;
+        case 64:
+          value = (isSigned) ? std::numeric_limits<int64_t>::max()
+                             : std::numeric_limits<uint64_t>::max();
+          break;
+        default:
           llvm_unreachable("unsupported element type");
         }
       })
       .Default([](Type) { llvm_unreachable("unsupported element type"); });
-  return rewriter.create<arith::ConstantOp>(loc, constantAttr);
+
+  MultiDialectBuilder<MathBuilder> create(rewriter, loc);
+  return create.math.constant(type, value);
 }
 
 Value emitNegativeInfinityConstantOp(
```

Comment:
This could be folded in the math, e.g. `math.negativeInfinity(type)` and `math.positiveInfinity(type)`.  Probably common enough a patterns for min/max kind of things.